protocol/consistencychecks.go

```diff
@@ -245,7 +245,11 @@ func (cc *ConsistencyChecks) updateTBs(requestType int, msg *Response,
 	case RegistrationType:
 		df := msg.DirectoryResponse.(*DirectoryProof)
 		if df.AP.ProofType() == m.ProofOfAbsence {
-			if err := cc.verifyReturnedPromise(df, key); err != nil {
+			tbKey := key
+			if msg.Error == ReqNameExisted {
+				tbKey = df.TB.Value
+			}
```

Comment:
We get `ErrBindingsDiffer` (either in `verifyReturnedPromise` or `verifyAuthPath`) iff `msg.Error == ReqNameExisted`, right? In other words, does `ReqNameExisted` always inconsistent with our expectation (assume that we never do registration twice for a user)? 
If so, do we want to fail this check?